Sources/PrimerSDK/Classes/User Interface/Apple Pay/ApplePayViewModel.swift

```diff
@@ -1,69 +1,317 @@
+        }
+        
+        guard let merchantIdentifier = merchantIdentifier else {
+            return completion(AppleException.missingMerchantIdentifier)
+        }
+        
+        guard !orderItems.isEmpty else {
+            return completion(PaymentException.missingOrderItems)
+        }
+        
+        let applePayRequest = ApplePayRequest(
+            currency: currency,
+            merchantIdentifier: merchantIdentifier,
+            countryCode: countryCode,
+//            supportedNetworks: supportedNetworks,
+            items: orderItems
+//            merchantCapabilities: merchantCapabilities
+        )
+        
+        var supportedNetworks: [PKPaymentNetwork] = [
+            .amex,
+            .chinaUnionPay,
+            .discover,
+            .interac,
+            .masterCard,
+            .privateLabel,
+            .visa
+        ]
+        
+        if #available(iOS 11.2, *) {
+//            @available(iOS 11.2, *)
+            supportedNetworks.append(.cartesBancaires)
+        } else if #available(iOS 11.0, *) {
+//            @available(iOS, introduced: 11.0, deprecated: 11.2, message: "Use PKPaymentNetworkCartesBancaires instead.")
+            supportedNetworks.append(.carteBancaires)
+        } else if #available(iOS 10.3, *) {
+//            @available(iOS, introduced: 10.3, deprecated: 11.0, message: "Use PKPaymentNetworkCartesBancaires instead.")
+            supportedNetworks.append(.carteBancaire)
+        }
+
+        if #available(iOS 12.0, *) {
+//            @available(iOS 12.0, *)
+            supportedNetworks.append(.eftpos)
+            supportedNetworks.append(.electron)
+            supportedNetworks.append(.maestro)
+            supportedNetworks.append(.vPay)
+        }
+
+        if #available(iOS 12.1.1, *) {
+//            @available(iOS 12.1.1, *)
+            supportedNetworks.append(.elo)
+            supportedNetworks.append(.mada)
+        }
+        
+        if #available(iOS 10.3.1, *) {
+//            @available(iOS 10.3, *)
+            supportedNetworks.append(.idCredit)
+        }
+        
+        if #available(iOS 10.1, *) {
+//            @available(iOS 10.1, *)
+            supportedNetworks.append(.JCB)
+            supportedNetworks.append(.suica)
+        }
+        
+        if #available(iOS 10.3, *) {
+//            @available(iOS 10.3, *)
+            supportedNetworks.append(.quicPay)
+        }
+        
+        if #available(iOS 14.0, *) {
+//            @available(iOS 14.0, *)
+//            supportedNetworks.append(.barcode)
+            supportedNetworks.append(.girocard)
+        }
+        
+        if PKPaymentAuthorizationViewController.canMakePayments(usingNetworks: supportedNetworks) {
+            let request = PKPaymentRequest()
+            request.currencyCode = applePayRequest.currency.rawValue
+            request.countryCode = applePayRequest.countryCode.rawValue
+            request.merchantIdentifier = "merchant.primer.dev.evangelos"
```

Comment:
[Feedback]
This shouldn't end up in the final code. I think the merchant identifier should come from the configuration?

Reply:
🤫
Fixed